prover/protocol/wizard/gnark_verifier.go

```diff
@@ -179,39 +179,42 @@ func (c *WizardVerifierCircuit) generateAllRandomCoins(_ frontend.API) {
 			toBeConsumed := c.Spec.Coins.AllKeysAt(currRound - 1)
 			c.Coins.Exists(toBeConsumed...)
 
-			// Make sure that all messages have been written and use them
-			// to update the FS state. Note that we do not need to update
-			// FS using the last round of the prover because he is always
-			// the last one to "talk" in the protocol.
-			toUpdateFS := c.Spec.Columns.AllKeysProofAt(currRound - 1)
-			for _, msg := range toUpdateFS {
-
-				msgID := c.columnsIDs.MustGet(msg)
-				msgContent := c.Columns[msgID]
-
-				logrus.Tracef("VERIFIER CIRCUIT : Updating the FS oracle with a message - %v", msg)
-				c.FS.UpdateVec(msgContent)
-			}
-
-			toUpdateFS = c.Spec.Columns.AllKeysPublicInputAt(currRound - 1)
-			for _, msg := range toUpdateFS {
-
-				msgID := c.columnsIDs.MustGet(msg)
-				msgContent := c.Columns[msgID]
-
-				logrus.Tracef("VERIFIER CIRCUIT : Updating the FS oracle with public input - %v", msg)
-				c.FS.UpdateVec(msgContent)
-			}
-
-			/*
-				Also include the prover's allegations for all evaluations
-			*/
-			queries := c.Spec.QueriesParams.AllKeysAt(currRound - 1)
-			for _, qName := range queries {
-				// Implicitly, this will panic whenever we start supporting
-				// a new type of query params
-				params := c.GetParams(qName)
-				params.UpdateFS(c.FS)
+			if !c.Spec.DummyCompiled {
```

Comment:
That condition had been removed from verifier.go and it was there only to ensure circuits are not too big when using the dummy compiler. That had an impact when testing, but we figured it is better to just not test circuit generation uusing dummy compilers anyway.